### ISSUE TITLE
Uploads bigger than 2 megabytes~?(I assume) fail to load the latter sets after the data limit.

### ISSUE THREAD

**Issue by DDDMEMAYBE** [Cadrach/lol_ise#33]
### Browser (Chrome, Safari, Internet Explorer) and Version
Chrome (Edit: Version 58.0.3029.81, updated version & issue was retained)
### Provide a Screenshot when possible
![capture1](https://cloud.githubusercontent.com/assets/28005704/25388513/15339f1e-299c-11e7-8fae-22deee317c4c.PNG)
![capture2](https://cloud.githubusercontent.com/assets/28005704/25388516/171e0e86-299c-11e7-952a-aa4592e46c39.PNG)
![capture3](https://cloud.githubusercontent.com/assets/28005704/25388522/1aa4c00e-299c-11e7-8546-1359e4f837fd.PNG)
![capture4](https://cloud.githubusercontent.com/assets/28005704/25388532/1d7753b4-299c-11e7-94d4-2c8710df6585.png)
![capture5](https://cloud.githubusercontent.com/assets/28005704/25388536/1fc41dd2-299c-11e7-9076-53db7c334e8b.png)
![capture6](https://cloud.githubusercontent.com/assets/28005704/25388540/2194ef56-299c-11e7-9ae1-1d6a1cfe610c.PNG)
![capture7](https://cloud.githubusercontent.com/assets/28005704/25388547/22c54678-299c-11e7-8a29-caf99037c77f.PNG)
![capture8](https://cloud.githubusercontent.com/assets/28005704/25388551/25525d90-299c-11e7-89d9-59a082e3690d.PNG)
![capture9](https://cloud.githubusercontent.com/assets/28005704/25388559/288a4928-299c-11e7-804c-f0bf79a75914.PNG)

### Steps to reproduce
Have a ton of .Json files. Easy to do if you have a lot of non-champion specified item sets due to how riot documents these. Each of those is granted a .json file for each champion, so if you have 50 of those like me. That's 50 x 136 files. Or somewhere around 6,750 bonus Json files to clog up the data. If you want to delete every item set, search "RIOT_ItemSet_" in your config folder(a copy of the config folder to be safe) and delete every item that appears in the search.
Note: The differently named item sets at the top of Capture2.png is from the "Lol Item Creator" I used, thinking it would let me create my itemsets. It just made more recommended item builds... They were somewhat useful so I kept them anyways though. I don't have anywhere near 30 item sets on Akali by the way. I have like 1 other than those 5 generated ones.

**Comment by Cadrach** [comment]
Ok, just so I am sure that I understand, there is no issue for me to fix right?

Currently, the tool is not limited by the file size (everything is read locally, nothing is uploaded anywhere so it is not a big issue), but there is a limit on the number of file read, around 3000 from memory (to avoid someone trying to read their C: drive for example, and crashing the browser or having to wait a loooong time)

**Comment by DDDMEMAYBE** [comment]
The way Riot's code handles unspecified-champion item kits makes it break this program. I don't know if there's anyway to fix it but you can't have un-specified champion item kits and use this program simultaneously, at least not more than five or so. It just needs a disclaimer perhaps for those kinds of item sets and how they are increasingly incompatible the more of them you have saved. The only way to fix this is to wait for the riot announced new item builder they have just announced or put the number limit to an absurd number. Perhaps make the limit in file size instead of number, since it's still going to be a small number of megabytes regardless. Maybe 6-8 or so Megabytes?

**Comment by Cadrach** [comment]
I don't think I can get the filesize without parsing all files (the file size in the end is the sum of the size of all files). If I can get your config folder (the big one) I can try and look at possible solutions.

**Comment by DDDMEMAYBE** [comment]
So you mean you can't change Chrome's allowed limit for _total number of files or total file size_?
I believe you are saying it's too laggy to search if any item set **is exactly the same** _and_ also has over 133 copies,(since there are 134~+ Champions, and "unspecified-champion item builds" are granted to every champion regardless of whether you own it or not.) The only theoretical solutions are: Parsing every file to see if there is any "amassed exactly congruent files" over 133~. Increasing the data limit. Creating a stand-alone download that fixes the issue directly to a specified folder or zipped folder(a tricky one).
**-->>>** Or, having a disclaimer or help note: to run the editor completely blank, download the item set you created while running the editor it blank/fresh, then add it to your champion's file folder in the respective riot games partition on your computer (remember you need to rename, or add a(1, 2, 3, ect) to, any sets for champions that already have item sets). I think that's the best current solution, using the editor blank if uploading doesn't work, and manually deleting any sets you don't want. **Another option is to only upload specific character's folders to decrease the file number.** **<<<--**
You also can't try and delete or ignore every "RIOT_ItemSet_1.json" past a certain number, obviously as some people may have 20 sets for one champion.

ps. Sorry for all the trouble here, I would be happy if you add a big note in help section saying that non-champion specified item sets are heavily bloated. That if you are missing champions in the editor, simply upload one champion's folder at a time or something, to the editor. Double click the config folder, go into champions and copy the champion you wish to change.
pss. I just tested around some more, If I upload the first 50 champions' folders, it gives me 17 champions, If I upload 134 champions, I get 42. For some reason there isn't an item set folder for Rakan or Xayah, even though I own Rakan and played him, and played with Xayah's in game(so there's 134 champions for item sets). I assume there isn't those 2 since they removed the legacy client and therefor item sets on the same patch as their release.

Github won't allow me to upload the zip file.

![capture](https://cloud.githubusercontent.com/assets/28005704/25468075/16923b40-2ae1-11e7-8151-aac0419b7df8.PNG)
![capture 1](https://cloud.githubusercontent.com/assets/28005704/25468076/174b46d0-2ae1-11e7-846a-06f4fbc4cb4c.PNG)

**Comment by Cadrach** [comment]
Ok, so:

- I think I can fix your issue, I just need your files to see the details of what might be wrong
- You can use something like wetransfer.com to send me the file if it is too big
- http://lol.item-set.com/#!/editor is indeed my site, this is how you got here right? I do not understand what you found dangerous there?

**Comment by DDDMEMAYBE** [comment]
The first reply post by you stated: "(everything is read locally, nothing is uploaded anywhere so it is not a big issue)"
You implied that this program build never uploaded anything, which had me, as a noob, think that implied the act of not-uploading anything was incompatible with the website's functions of accepting my data. Leading me to believe it was impossible for the website to be of your volition. The website also says: "Upload" to import item sets, probably for noobs like me to understand. So, yea- at least my mis-comprehension had a basis as I was coming from both the website's wording and your wordings. Again, sorry for the confusion. I'll try to upload my file sets again.

**Comment by DDDMEMAYBE** [comment]
I'm just going to upload the champions folder as it should say enough.? I don't feel like skimming through the .json files of the config folder out of schizophrenia again lol. We transfer needs an email address to send it to you; I'm too braindead to find your email on github so bear with me if you want the stupid folder lolol. I tried 7zip and some classical zipping processes but github won't accept either.

**Comment by Cadrach** [comment]
For wetransfer you can use your own email address, this way you will get the download link which is public and you can paste it here!

**Comment by DDDMEMAYBE** [comment]
After looking through the files again, I realized there was only this problem for champions I had created item sets for, or perhaps it's those that I've never played in game?, It would have none of the large sum of sets for champions that I didn't play/make a build with.

https://we.tl/d6AywuGcKK   // WeTransfer link.

I really hope this isn't some third party bs after all this. I wish I could make universal sets again in client and test but I can't. It's probably riot from seeing other responses but I can't discern.
